testing/testing.go

```diff
@@ -186,13 +184,13 @@ go.exe run GO_SCRIPT_PATH %*
 	}
 }
 
-// RunGitServer starts serving git HTTP server and returns its address including port
-func RunGitServer(t *testing.T, gitRoot string) (hostPort string) {
-	l, err := net.Listen("tcp", "localhost:0")
+// RunGitServer starts serving git HTTP server and returns its address
+func RunGitServer(root string, t *testing.T) (url string) {
```

Comment:
Returning the full HTTP URL to the server's root seemed more obvious what is happening.